content/doc/tutorials/using-jenkinsfile-runner-github-action-to-build-jenkins-pipeline.adoc

```diff
@@ -0,0 +1,308 @@
+If you want to learn more details about viewing your workflow details, you can check link:https://docs.github.com/en/actions/quickstart#viewing-your-workflow-results[the official guide].
+[.boxshadow]
+image:tutorials/jenkinsfile-runner-github-actions-01-access-workflow-logs.jpeg[alt="Access the workflow logs",width=100%]
+
+=== Add JCasC (Optional)
+
+Usually we need to access the web UI to set up Jenkins. 
+However, we’re unable to access the web UI under the circumstances of running Jenkins pipeline in the GitHub Actions because the Jenkins instance is ephemeral. 
+JCasC (link:https://github.com/jenkinsci/configuration-as-code-plugin[Jenkins Configuration as Code]) plugin can configure this ephemeral Jenkins instance by providing the human-readable declarative configuration files. 
+
+In this example, we show how to set up the environment variables by JCasC and access them in the Jenkinsfile.
+* Create a JCasC YAML file called jcasc.yml and declare the environment variables:
+[source,yaml]
+----
+jenkins:
+  globalNodeProperties:
+    - envVars:
+        env:
+          - key: hello
+            value: world
+----
+* Create a Jenkinsfile:
+[source,groovy]
+----
+pipeline {
+    agent any
+    stages {
+        stage('test casc env') {
+            steps {
+                echo "JCasC env.hello: ${env.hello}"
+            }
+        }
+    }
+}
+----
+* Specify the jcasc.yml in the GitHub Actions input:
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+----
+
+You can find more link:https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos[examples] provided by configuration-as-code-plugin and learn how to configure the Jenkins instance without using the UI page. 
+Some plugins might not have concrete examples but you can debug and find their JCasC in the UI page. 
+You can check the configuration in Manage Jenkins -> Configuration as Code -> View Configuration. 
+Then you can copy the parts you need to the JCasC file.
+
+=== Add and configure some plugins (Optional)
+
+Jenkins has a powerful plugin system. 
+You can add the plugins in the plugin list file and configure the plugins in the JCasC YAML file as needed.
+
+In this example, we show how to install JDK8 in the ephemeral Jenkins instance.
+* Specify `adoptopenjdk` plugin in the plugins.txt file. 
+As the version is not specified, the latest version will be installed.
+[source,txt]
+----
+adoptopenjdk
+----
+* Create a JCasC Yaml file called jcasc.yml and specify which JDK version to install.
+[source,yaml]
+----
+tool:
+  jdk:
+    installations:
+      - name: jdk8
+        home: "~/jdk8"
+        properties:
+          - installSource:
+              installers:
+                - adoptOpenJdkInstaller:
+                    id: "jdk8u332-b09"
+----
```

Comment:
```suggestion
[source,yaml]
----
tool:
  jdk:
    installations:
      - name: jdk8
        home: "~/jdk8"
        properties:
          - installSource:
              installers:
                - adoptOpenJdkInstaller:
                    id: "jdk8u332-b09"
----
```
can jdk11 be used here?

Reply:
I think it can work. Let me test it :)

Reply:
I just changed the guide to use jdk11. Thanks again for your advice!